File: dlibs/low-utils/lib/src/utils/grcpRequest.dart
import 'package:grpc/grpc.dart' as grpc;

typedef AsyncGetter<TRepply> = Future<TRepply> Function(
    grpc.ClientChannel channel);

Future<TRepply> grpcRequest<TRepply>(AsyncGetter<TRepply> getter, int port, String host) async {
  final channel = grpc.ClientChannel(host,
      port: port,
      options: const grpc.ChannelOptions(
          credentials: const grpc.ChannelCredentials.insecure()));
  TRepply message;
  try {
    message = await getter(channel);
  } catch (e) {
    print('Caught error: $e');
  }
  await channel.shutdown();
  return message;
}

File: clibs/messages/grcpRequest.cs
using Grpc.Core;
using System;
using System.Threading.Tasks;

namespace RewiseDom {
  public static class Client {
    public static TRepply grpcRequest<TRepply>(Func<Channel, TRepply> getter, int port, String host) {
      var channel = new Channel(host, port, ChannelCredentials.Insecure);
      try {
        var msg = getter(channel);
        return msg;
      } finally {
        channel.ShutdownAsync().Wait();
      }
    }
  }
}